backend/compact-connect/lambdas/provider-data-v1/tests/function/test_handlers/test_ingest.py

```diff
@@ -0,0 +1,252 @@
+        # Removing dynamic fields from comparison
+        del expected_provider['providerId']
+        del provider_data['providerId']
+        del expected_provider['dateOfUpdate']
+        del provider_data['dateOfUpdate']
+        del expected_provider['licenses'][0]['dateOfUpdate']
+        del provider_data['licenses'][0]['dateOfUpdate']
+
+        self.assertEqual(
+            expected_provider,
+            provider_data
+        )
+
+    def test_old_inactive_license(self):
+        from handlers.ingest import ingest_license_message
+        from handlers.providers import get_provider
+
+        # The test resource provider has a license in oh
+        self._load_provider_data()
+        with open('tests/resources/dynamo/provider-ssn.json', 'r') as f:
+            provider_id = json.load(f)['providerId']
+
+        with open('tests/resources/ingest/message.json', 'r') as f:
+            message = json.load(f)
+        # Imagine that this provider used to be licensed in ky.
+        # What happens if ky uploads that inactive license?
+        message['detail']['dateOfIssuance'] = '2023-01-01'
+        message['detail']['familyName'] = 'Oldname'
+        message['detail']['jurisdiction'] = 'ky'
+        message['detail']['status'] = 'inactive'
+
+        event = {
+            'Records': [
+                {
+                    'messageId': '123',
+                    'body': json.dumps(message)
+                }
+            ]
+        }
+
+        resp = ingest_license_message(event, self.mock_context)  # pylint: disable=too-many-function-args
+
+        self.assertEqual(
+            {'batchItemFailures': []},
+            resp
+        )
+
+        # To test full internal consistency, we'll also pull this new license record out
+        # via the API to make sure it shows up as expected.
+        with open('tests/resources/api-event.json', 'r') as f:
+            event = json.load(f)
+
+        event['pathParameters'] = {'compact': 'aslp', 'providerId': provider_id}
+        event['requestContext']['authorizer']['claims']['scope'] = 'openid email stuff aslp/read'
+        resp = get_provider(event, self.mock_context)
+        self.assertEqual(resp['statusCode'], 200)
+
+        with open('tests/resources/api/provider-detail-response.json', 'r') as f:
+            expected_provider = json.load(f)
+
+        provider_data = json.loads(resp['body'])
+
+        # Removing dynamic fields from comparison
+        del expected_provider['providerId']
+        del provider_data['providerId']
+        del expected_provider['dateOfUpdate']
+        del provider_data['dateOfUpdate']
+
+        # We will look at the licenses separately
+        del expected_provider['licenses']
+        licenses = provider_data.pop('licenses')
+
+        # The original provider data is preferred over the posted license data in our test case
+        self.assertEqual(
+            expected_provider,
+            provider_data
+        )
```

Comment:
Not just their privileges. We are also making an attempt to save personal information (name, address, DOB, etc) about the provider, based on our best guess of the most reliable data source (among various licenses). In the provider list and detail views, we present their given and family name among other data. Because that information can change in a person's lifetime, it is possible that different jurisdictions may have different names for the same individual. To mitigate this, we take our best guess of what license to consider the most reliable and save those values from the license onto the provider, for presentation on the providers table and detail view. In this case, the license we ingested is considered less reliable than the one we already have loaded, so the provider name, etc should remain unchanged. We should, however, still see the license added to their license list, as you said.

Reply:
got it!